src/util_i_variables.nss

```diff
@@ -0,0 +1,2051 @@
+                    "tag = @tag, " +
+                    "timestamp = strftime('%s','now');";
+
+    sqlquery q = bPC || !bCampaign ? _PrepareQueryObject(oObject, s) : _PrepareQueryCampaign(s);
+    SqlBindInt   (q, "@type", nType);
+    SqlBindString(q, "@varname", sVarName);
+    SqlBindString(q, "@tag", sTag);
+    return q;
+}
+
+sqlquery _PrepareTagUpdate(object oObject, int nType, string sVarName, string sTag, int bCampaign)
+{
+    int bPC = GetIsPC(oObject);
+    string sTable = bPC ? VARIABLE_TABLE_PC : VARIABLE_TABLE_MODULE;
+
+    string s =  "UPDATE " + sTable + " " +
+                "SET tag = @tag " +
+                "WHERE type = @type " +
+                    "AND varname = @varname;";
+
+    sqlquery q = bPC || !bCampaign ? _PrepareQueryObject(oObject, s) : _PrepareQueryCampaign(s);
+    SqlBindInt   (q, "@type", nType);
+    SqlBindString(q, "@varname", sVarName);
+    SqlBindString(q, "@tag", sTag);
+    return q;    
+}
+
+/// @brief Prepares an delete query to remove a variable stored in any database.
+sqlquery _PrepareSimpleVariableDelete(object oObject, int nType, string sVarName, int bCampaign)
+{
+    int bPC = GetIsPC(oObject);
+    string sTable = bPC ? VARIABLE_TABLE_PC : VARIABLE_TABLE_MODULE;
+
+    string s =  "DELETE FROM " + sTable + " " +
+                "WHERE type = @type " +
+                    "AND varname = @varname " +
+                "RETURNING value;";
+
+    sqlquery q = bPC || !bCampaign ? _PrepareQueryObject(oObject, s) : _PrepareQueryCampaign(s);
+    SqlBindInt   (q, "@type", nType);
+    SqlBindString(q, "@varname", sVarName);
+    return q;
+}
+
+/// @brief Prepares a complex delete query to remove multiple variables by criteria.
+/// @param nType Bitwise VARIABLE_TYPE_*
+/// @param sVarName Variable name pattern, accept glob patterns, sets and wildcards
+/// @param sTag Tag pattern, accepts glob patterns, sets and wildcards
+/// @param nTime (+) is after, (-) is before
+/// @warning If no parameters are passed, this query will result in a simple "DELETE ALL"
+///     and will delete all variables in oObject's database.
+sqlquery _PrepareComplexVariableDelete(object oObject, int nType = VARIABLE_TYPE_NONE, string sVarName = "", 
+                                       string sTag = "", int nTime = 0, int bCampaign = FALSE)
+{
+    int n, bPC = GetIsPC(oObject);
+    string sTable = bPC ? VARIABLE_TABLE_PC : VARIABLE_TABLE_MODULE;
+    string sWhere =  (sVarName == "" ? "" : " $" + IntToString(++n) + " varname GLOB @varname");
+           sWhere += (sTag == ""     ? "" : " $" + IntToString(++n) + " tag GLOB @tag");
+           sWhere += (nType <= 0     ? "" : " $" + IntToString(++n) + " type & @type > 0");
+           sWhere += (nTime == 0     ? "" : " $" + IntToString(++n) + " timestamp " + (nTime > 0 ? ">" : "<") + " @time");
+    
+    json jKeyWords = ListToJson("WHERE,AND,AND,AND");
+    string s = "DELETE FROM " + sTable + sWhere + ";";
+           s = SubstituteString(s, jKeyWords);
```

Comment:
Heh, I was oddly proud of this function.  Not sure why, really.  Just kept re-formatting it because I was so annoyed with the possibility of having to use ternaries to determine if the where clause was already populated.  When I first typed `sWhere += (sWhere == "" : " WHERE " : " AND ") + (sVarName == "" ? "" : " varname GLOB @varname")`, my first reaction was "absolutely not".  I'm sure there's still more optimizations available here, I'll keep an eye out.

Reply:
Also, any thoughts on setting timestamps based on your time include versus unix time?  Unix time might generally be more easily accessible and can be accessed via sqlite, but would also be difficult to convert into in-game times if that's necessary for any reason.

Reply:
Nah, keeping the timestamps based on sqlite functions is the way to go.